lib/js/components/TextGroup/TextGroup.vue

```diff
@@ -0,0 +1,296 @@
+		:class="{
+			'-ds-small': size === TEXT_GROUP_SIZES.SMALL,
+
+			'-ds-hovered': state === TEXT_GROUP_STATES.HOVERED,
+			'-ds-loading': isLoading,
+			'-ds-disabled': state === TEXT_GROUP_STATES.DISABLED,
+
+			'-ds-interactive': isInteractive,
+			'-ds-selected': isSelected,
+
+			[loadingSizeClassName]: isLoading,
+
+			[colorClassName]: true,
+		}"
+	>
+		<div
+			v-if="eyebrowText !== null"
+			class="ds-textGroup__eyebrow"
+			:class="{
+				'-ds-ellipsis': eyebrowTextEllipsis,
+				'-ds-uppercase': isEyebrowTextUppercase,
+			}"
+		>
+			<ds-skeleton v-if="isLoading" width="50%" height="100%" />
+			<span v-else>{{ eyebrowText }}</span>
+		</div>
+		<div
+			v-if="mainText !== null"
+			class="ds-textGroup__main"
+			:class="{
+				'-ds-ellipsis': mainTextEllipsis,
+			}"
+		>
+			<ds-skeleton v-if="isLoading" width="100%" height="100%" />
+			<span v-else>{{ mainText }}</span>
+		</div>
+		<div
+			v-if="supportingText !== null"
+			class="ds-textGroup__supporting"
+			:class="{
+				'-ds-ellipsis': supportingTextEllipsis,
+			}"
+		>
+			<ds-skeleton v-if="isLoading" width="100%" height="100%" />
+			<span v-else v-html="supportingText" />
+		</div>
+	</div>
+</template>
+
+<style lang="scss">
+@import '../../../styles/settings/typography/tokens';
+@import '../../../styles/settings/colors/tokens';
+@import '../../../styles/settings/spacings';
+
+.ds-textGroup {
+	$self: &;
+
+	display: flex;
+	flex-direction: column;
+	width: 100%;
+
+	&__eyebrow {
+		@include info-s-default-bold;
+
+		color: $color-neutral-text-weak;
+		margin-bottom: $space-5xs;
+		min-height: 12px;
+
+		&.-ds-uppercase {
+			@include info-s-extensive-bold-uppercase;
+		}
+	}
+
+	&__main {
+		@include text-m-compact-bold;
+
+		color: $color-neutral-text-strong;
+		min-height: 18px;
+	}
+
+	&__supporting {
+		@include text-s-compact-regular;
+
+		color: $color-neutral-text;
+		min-height: 16px;
+	}
+
+	&__eyebrow.-ds-ellipsis,
+	&__main.-ds-ellipsis,
+	&__supporting.-ds-ellipsis {
+		overflow: hidden;
+		text-overflow: ellipsis;
+		white-space: nowrap;
+	}
+
+	&.-ds-small {
+		#{$self}__eyebrow {
+			@include info-xs-default-bold;
+
+			margin-bottom: 0;
+			min-height: 10px;
+
+			&.-ds-uppercase {
+				@include info-xs-extensive-bold-uppercase;
+			}
+		}
+
+		#{$self}__main {
+			@include text-s-compact-bold;
+
+			min-height: 16px;
+		}
+	}
+
+	&.-ds-interactive:hover,
+	&.-ds-hovered {
+		#{$self}__eyebrow {
+			color: $color-neutral-text-weak-hovered;
+		}
+
+		#{$self}__main {
+			color: $color-neutral-text-strong-hovered;
+		}
+
+		#{$self}__supporting {
+			color: $color-neutral-text-hovered;
+		}
+	}
+
+	&.-ds-disabled.-ds-interactive:hover,
+	&.-ds-disabled {
+		#{$self}__eyebrow {
+			color: $color-neutral-text-weak-disabled;
+		}
+
+		#{$self}__main {
+			color: $color-neutral-text-strong-disabled;
+		}
+
+		#{$self}__supporting {
+			color: $color-neutral-text-disabled;
+		}
+	}
+
+	&.-ds-disabled.-ds-selected.-ds-interactive:hover,
```

Comment:
usually, we set `pointer-events: none` on disabled elements, so there is no hover state. Is it different here?